mkt/extensions/views.py

```diff
@@ -58,7 +57,9 @@ def create(self, request, *args, **kwargs):
         if not file_obj:
             raise exceptions.ParseError(_('Missing file in request.'))
 
-        self.validate_upload(file_obj)  # Will raise exceptions if appropriate.
+        # Will raise exceptions if appropriate.
+        ExtensionValidator(file_obj).validate()
```

Comment:
At some point we'll have to take care of the `FIXME` a few lines below, or at least not forget it once we run the validation asynchronously, but I can't think of a good way to integrate that with your changes right now.